src/braket/pennylane_plugin/ahs_device.py

```diff
@@ -0,0 +1,453 @@
+        else:
+            vals = [pulse_parameter for t in time_points]
+
+        for t, v in zip(time_points, vals):
+            ts.put(t, v)
+
+        return ts
+
+    def _convert_pulse_to_driving_field(self, pulse, time_interval):
+        """Converts a ``HardwarePulse`` from PennyLane describing a global drive to a 
+        ``DrivingField`` from Braket AHS
+
+        Args:
+            pulse[HardwarePulse]: a dataclass object containing amplitude, phase and detuning 
+                information
+            time_interval(array[Number, Number]]): The start and end time for the applied pulse
+
+        Returns:
+            DrivingField: the object representing the global drive for the
+                AnalogueHamiltonianSimulation object
+        """
+
+        time_points = self._get_sample_times(time_interval)
+
+        # scaling factor for amp and detuning converts MHz (PL input) to rad/s (upload units)
+        amplitude = self._convert_to_time_series(
+            pulse.amplitude, time_points, scaling_factor=2 * np.pi * 1e6
+        )
+        detuning = self._convert_to_time_series(
+            pulse.detuning, time_points, scaling_factor=2 * np.pi * 1e6
+        )
+        phase = self._convert_to_time_series(pulse.phase, time_points)
+
+        drive = DrivingField(amplitude=amplitude, detuning=detuning, phase=phase)
+
+        return drive
+
+    @staticmethod
+    def _result_to_sample_output(res):
+        """This function converts a single shot of the QuEra measurement results to
+        0 (ground), 1 (excited) and NaN (failed to measure) for all atoms in the result.
+
+        The QuEra results are summarized via 3 values: status, pre_sequence, and post_sequence.
+
+        Status is success or fail. The pre_sequence is 1 if an atom in the ground state was
+        successfully initialized, and 0 otherwise. The post_sequence is 1 if an atom in the
+        ground state was measured, and 0 otherwise. Comparison of pre_sequence and post_sequence
+        reveals one of 3 possible outcomes:
+
+        0 --> 0: Atom failed to be placed (no atom detected in the ground state either before or after)
+        1 --> 0: Atom initialized, measured in Rydberg state (atom in ground state detected before, but not after)
+        1 --> 1: Atom initialized, measured in ground state (atom in ground state detected both before and after)
+        """
+
+        # if entire measurement failed, all NaN
+        if not res.status.value.lower() == "success":
+            return np.array([np.NaN for i in res.pre_sequence])
+
+        # if a single atom failed to initialize, NaN for that individual measurement
+        pre_sequence = [i if i else np.NaN for i in res.pre_sequence]
+
+        # set entry to 0 if ground state is measured, 1 if excited state is measured, NaN if measurement failed
+        return np.array(pre_sequence - res.post_sequence)
+
+
+class BraketAquilaDevice(BraketAhsDevice):
```

Comment:
Rather than name the QPU directly, use a more generic name like `BraketAwsAhsDevice`